Mathlib/Data/List/GroupBy.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2024 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Data.List.Chain
+
+/-!
+# Group a list by a relation
```

Comment:
Can you clarify in the module doc that it is grouping *consecutive* elements? I was expecting `groupBy (. = .) [1, 2, 1] = [[1, 1], [2]]`